### PR TITLE
GraalJS parser JAR removed from distribution. Created convenience mod…

### PR DESCRIPTION
**Pull request by sdedic**
…ule to produce distribution NBM for non-Apache distributors.

The purpose of this PR is to satisfy distribution restrictions of Apache while assuring NB users will get the javascript editing features.

JS editing in NetBeans relies on code evolved from oracle JS parser (see http://hg.openjdk.java.net/graal/graal-js-parser). The code is licensed using `GPL with classpath exception` and must not be distributed in a convenience binary by Apache.

At the same time, the license allows to distribute the parser's code for other parties. So I changed the original wrapper module to use the parser's JAR at compile time only; the JAR will not be copied to the build output. 
Then, I provided an additional "convenience" module, which **is not part of any cluster or any NB build** which will produce a `NBM` that contais the JS parser code for distribution. 

NetBeans "feature on demand" support was changed so that it knows that an additional module is required and will attempt to locate the module on Update Centers (namely plugin portal). That means that if the NBM for the convenience module appears on Plugin Portal, it will be downloaded during activation and JS support will start to work.

During the course I also fixed some minor issues:

- if update fetch fails with IOException, no error dialog will show, errors will be printed into the wizard panel
- if the required "extra" module is not available, wizard will block activation procedure

Since the JS module **is not on the Plugin Portal now**, I put a testing module to my personal webspace - for **testing** use http://bordel.klfree.net/~belgarat/nb/pp/catalog.xml as an additional update catalog

### CODE REVIEW

File: webcommon/libs.nashorn/build.xml [hunk 1]
```diff
@@ -22,4 +22,10 @@
 <project basedir="." default="build" name="webcommon/libs.nashorn">
     <description>Builds, tests, and runs the project org.netbeans.libs.nashorn</description>
     <import file="../../nbbuild/templates/projectized.xml"/>
+    <import file="../../nbbuild/templates/common.xml"/>
+
+    <!-- For testing purposes, build distribution of JS parser in a separate directory -->
```
Comment (JaroslavTulach) [inline webcommon/libs.nashorn/build.xml]: Unzipping it into `test/classes` directory might make it available to others via standard `<test>` dependency.

File: webcommon/libs.nashorn/src/org/netbeans/libs/nashorn/Bundle.properties [hunk 2]
```diff
@@ -14,4 +14,4 @@
 # KIND, either express or implied.  See the License for the
 # specific language governing permissions and limitations
 # under the License.
-OpenIDE-Module-Name=Nashorn Integration
+OpenIDE-Module-Name=Nashorn Integration
```
Comment (JaroslavTulach) [inline webcommon/libs.nashorn/src/org/netbeans/libs/nashorn/Bundle.properties]: Missing new line.

**Comment by sdedic** [comment]
sorry, still some leftovers on local disk :(

**Comment by sdedic** [comment]
Colleagues: the JS Parser implementation NBM is live on UC for NB10; last chance to test this entire feature before the PR is merged. I have rebased on latest master which changes AUC of master to UC10, so refresh your sources before testing.

**Comment by geertjanw** [comment]
What must be done to try out the feature? If you provide step by step instructions, I'd be happy to provide feedback.

**Comment by sdedic** [comment]
You need a build from the branch' sources. See the cmdline instructions:
`git checkout -b sdedic-jsparser_download master
git pull git://github.com/sdedic/incubator-netbeans.git jsparser_download
`
Then build and run.
Try to create a HTML5/JS project, or create a JS file.

**Comment by JaroslavTulach** [comment]
I tried to create new HTML or Node.js project and both things work for me.

**Comment by sdedic** [comment]
Rebased on latest master & squashed

**Comment by lkishalmi** [comment]
Cherry picked to release100

**Comment by matthiasblaesing** [comment]
@sdedic could you please revisit this? It breaks using the `full` cluster. I saw the following problem in master and bisected it to this commit:

0. start with a clean installation: `git clean -f -x -d`
1. Build the netbeans cluster with the `full` configuration: `ant -Dcluster.config=full`
2. Run `tryme`: `ant tryme`
3. Go to Plugin configuration - Menu: Tools -> Plugins
4. Switch to `Installed` tab - you see, that HTML5 is not active (expected)
5. Select the entry and choose `Activate`

Normally at this point several modules will be activated and afterwards the IDE can access the HTML5 features. Beginning with this commit various javascript modules fail to load (not sure if it is random which module fails, but I saw different names). Here is the full log from the session:

[3357fa5b18590908ac9de4ad7e55ab1fdf88cd50.log](https://github.com/apache/incubator-netbeans/files/2612462/3357fa5b18590908ac9de4ad7e55ab1fdf88cd50.log)

It would be great if you could fix this.

**Comment by lkishalmi** [comment]
Just tested. It also breaks release100 branch. The HTML5 feature cannot be activated any more.

**Comment by matthiasblaesing** [comment]
This is still broken in master. We might even release NetBeans 10.0 with this regression. @sdedic could you please have a look at this? If I'm not mistaken the graaljs parser (or even the whole codebase) was converted to UPL, so we could even include it, without downloading it at runtime.

**Comment by matthiasblaesing** [comment]
Sorry for the noise: I though I was on master, but I had a PR checked out, that was based on an old state of the master branch. I checked again with master and it works there. Sorry and merry christmas!

**Pull request merged**